src/include/miopen/graphapi/execution_plan.hpp

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/graphapi/enginecfg.hpp>
+#include <miopen/graphapi/variant_pack.hpp>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace miopen {
+
+namespace graphapi {
+
+class ExecutionPlan
+{
+private:
+    /* we don't use a pointer for mEngineCfg
+     * because we need to support serialization
+     * and deserialization
+     */
+    EngineCfg mEngineCfg;
```

Comment:
same here. Should be a `pointer`.

Reply:
Then we will not be able to deserialize it

Reply:
This place requires some managed pointer, shared_ptr, I think. Should we then switch to `shared_ptr`s throughout the existing graph API code?